packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show ParagraphBuilder;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+
+import 'basic_types.dart';
+import 'text_style.dart';
+import 'text_painter.dart';
+
+/// An immutable span of inline content which forms a paragraph.
+///
+/// The subclass [TextSpan] specifies text and may contain child [InlineSpan]s.
+///
+/// The subclass [PlaceholderSpan] represents a placeholder that may be
+/// filled with non-text content. [PlaceholderSpan] itself defines a
+/// [ui.PlaceholderAlignemnt] and a [TextBaseline]. To be useful,
+/// [PlaceholderSpan] should be extended to define content. An instance of
+/// this is the [WidgetSpan] class in the widgets library.
+///
+/// The subclass [WidgetSpan] specifies embedded inline widgets. Specify a
+/// widget by wrapping the widget with a [WidgetSpan]. The widget will be
+/// laid out inline within the paragraph.
```

Comment:
the second sentence here is a bit redundant with the others. i think we could remove it without making the text any less clear.